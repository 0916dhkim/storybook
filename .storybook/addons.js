// Uncomment to register defaults
// import '@kadira/storybook/addons';

// Use the line below to register this addon
// import '@kadira/storybook-addon-comments/register';
import '@kadira/storybook-database-cloud/register';
import { init } from '../src/manager';
init();
