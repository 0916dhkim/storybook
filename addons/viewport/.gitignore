*.sw*
node_modules
dist
jest
jest_0
