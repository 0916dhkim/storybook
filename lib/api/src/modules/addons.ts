import { Module } from '../index';
import { RenderData as RouterData } from '@storybook/router';
import { ReactElement } from 'react';

export enum types {
  TAB = 'tab',
  PANEL = 'panel',
  TOOL = 'tool',
  PREVIEW = 'preview',
  NOTES_ELEMENT = 'notes-element',
}

export type Types = types | string;

export interface Addon {
  title: string;
  type?: Types;
  id?: string;
  route?: (routeOptions: RouterData) => string;
  match?: (matchOptions: RouterData) => boolean;
  render: (renderOptions: RouterData) => ReactElement<any>;
}
export interface Collection {
  [key: string]: Addon;
}

interface Panels {
  [id: string]: Addon;
}

export interface SubAPI {
  getElements: (type: Types) => Collection;
  getPanels: () => Collection;
  getSelectedPanel: () => string;
  setSelectedPanel: (panelName: string) => void;
}

export function ensurePanel(panels: Panels, selectedPanel?: string, currentPanel?: string) {
  const keys = Object.keys(panels);

  if (keys.indexOf(selectedPanel) >= 0) {
    return selectedPanel;
  }

  if (keys.length) {
    return keys[0];
  }
  return currentPanel;
}

export default ({ provider, store }: Module) => {
  const api: SubAPI = {
    getElements: (type: Types) => provider.getElements(type),
    getPanels: () => api.getElements(types.PANEL),
    getSelectedPanel: () => {
      const { selectedPanel } = store.getState();
      return ensurePanel(api.getPanels(), selectedPanel, selectedPanel);
    },
    setSelectedPanel: (panelName: string) => {
      store.setState({ selectedPanel: panelName }, { persistence: 'session' });
    },
  };

  return {
    api,
    state: {
      selectedPanel: ensurePanel(api.getPanels(), store.getState().selectedPanel),
    },
  };
};
