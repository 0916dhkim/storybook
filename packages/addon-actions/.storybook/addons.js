// Use the line below to register this addon
// import '@kadira/storybook-addon-actions/register';
import '../register';
