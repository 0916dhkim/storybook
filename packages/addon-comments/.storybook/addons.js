// Use the line below to register this addon
// import '@kadira/storybook-addon-comments/register';
import '@kadira/storybook-addon-actions/register';
import '@kadira/storybook-database-cloud/register';
import { init } from '../src/manager';
init();
