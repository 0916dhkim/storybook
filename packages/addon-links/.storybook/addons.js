// Use the line below to register this addon
// import '@storybook/storybook-addon-links/register';
import '../register';
