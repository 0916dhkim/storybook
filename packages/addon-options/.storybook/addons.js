// Use the line below to register this addon
// import '@kadira/storybook-addon-hello/register';
import '../register';
