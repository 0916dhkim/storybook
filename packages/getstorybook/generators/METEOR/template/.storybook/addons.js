import '@storybook/addon-actions/register';
import '@storybook/storybook-addon-links/register';
