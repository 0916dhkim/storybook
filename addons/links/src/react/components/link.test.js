import { shallow } from 'enzyme';
import React from 'react';
import addons from '@storybook/addons';

import { SELECT_STORY } from '@storybook/core-events';
import LinkTo from './link';

jest.mock('@storybook/addons');

const mockChannel = () => {
  return {
    emit: jest.fn(),
    on: jest.fn(),
    once: jest.fn(),
  };
};

describe('LinkTo', () => {
  describe('render', () => {
    it('should render a link', async () => {
      const channel = mockChannel();
      addons.getChannel.mockReturnValue(channel);

      const wrapper = shallow(<LinkTo kind="foo" story="bar" />);
      await wrapper.instance().updateHref(wrapper.props());
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it('should select the kind and story on click', () => {
      const channel = {
        emit: jest.fn(),
        on: jest.fn(),
      };
      addons.getChannel.mockReturnValue(channel);

      const wrapper = shallow(<LinkTo kind="foo" story="bar" />);
      wrapper.simulate('click', { button: 0, preventDefault: () => {} });
      expect(channel.emit.mock.calls).toContainEqual([
        SELECT_STORY,
        {
          kind: 'foo',
          story: 'bar',
        },
      ]);
    });
  });
});
