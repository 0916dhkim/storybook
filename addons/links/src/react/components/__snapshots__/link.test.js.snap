// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkTo render should render a link 1`] = `
<RoutedLink
  href="http://localhost/?id=?selectedKind=foo&selectedStory=undefined"
  onClick={[Function]}
/>
`;
